core/winnt/src/TWinNTSystem.cxx

```diff
@@ -2256,20 +2256,27 @@ const char *TWinNTSystem::TempDirectory() const
 /// Create a secure temporary file by appending a unique
 /// 6 letter string to base. The file will be created in
 /// a standard (system) directory or in the directory
-/// provided in dir. The full filename is returned in base
+/// provided in dir. Optionally one can provide suffix
+/// append to the final name - like extension ".txt" or ".html".
+/// The full filename is returned in base
 /// and a filepointer is returned for safely writing to the file
 /// (this avoids certain security problems). Returns 0 in case
 /// of error.
 
-FILE *TWinNTSystem::TempFileName(TString &base, const char *dir)
+FILE *TWinNTSystem::TempFileName(TString &base, const char *dir, const char *suffix)
 {
    char tmpName[MAX_PATH];
 
    ::GetTempFileName(dir ? dir : TempDirectory(), base.Data(), 0, tmpName);
    base = tmpName;
-   FILE *fp = fopen(tmpName, "w+");
+   if (suffix && *suffix) {
+      TString base_suffix = base + suffix;
+      ::rename(base.Data(), base_suffix.Data());
```

Comment:
I don't think that is safe enough since `rename` would overwrite a potential target. The probability is low, but the impact potentially severe.

I think we need to use `CreateFile` with the `CREATE_NEW` flag, so that it fails if the file exists. Once we have the Windows file handle, we can use `_open_osfhandle` to turn it into a C file stream.